In [6]:
import pandas as pd
import seaborn as sb
import numpy as np
import matplotlib.pyplot as plt
import sklearn as sk

In [7]:
banking_dataset_path = "dataset/banking.csv"
df = pd.read_csv(banking_dataset_path)
df.head(5)

age          job  marital          education  default housing loan  \
0   44  blue-collar  married           basic.4y  unknown     yes   no   
1   53   technician  married            unknown       no      no   no   
2   28   management   single  university.degree       no     yes   no   
3   39     services  married        high.school       no      no   no   
4   55      retired  married           basic.4y       no     yes   no   

    contact month day_of_week  ...  campaign  pdays  previous     poutcome  \
0  cellular   aug         thu  ...         1    999         0  nonexistent   
1  cellular   nov         fri  ...         1    999         0  nonexistent   
2  cellular   jun         thu  ...         3      6         2      success   
3  cellular   apr         fri  ...         2    999         0  nonexistent   
4  cellular   aug         fri  ...         1      3         1      success   

  emp_var_rate  cons_price_idx  cons_conf_idx  euribor3m  nr_employed  y  
0          1.4          93.444          -36.1      4.963       5228.1  0  
1         -0.1          93.200          -42.0      4.021       5195.8  0  
2         -1.7          94.055          -39.8      0.729       4991.6  1  
3         -1.8          93.075          -47.1      1.405       5099.1  0  
4         -2.9          92.201          -31.4      0.869       5076.2  1  

[5 rows x 21 columns]

In [15]:
print(df.shape)
print(list(df.columns))

(41188, 21)
['age', 'job', 'marital', 'education', 'default', 'housing', 'loan', 'contact', 'month', 'day_of_week', 'duration', 'campaign', 'pdays', 'previous', 'poutcome', 'emp_var_rate', 'cons_price_idx', 'cons_conf_idx', 'euribor3m', 'nr_employed', 'y']


## Bemeneti paraméterek értelmezése
1.age (numeric) <br>
2.job : type of job (categorical: “admin”, “blue-collar”, “entrepreneur”, “housemaid”, “management”, “retired”, “self-employed”, “services”, “student”, “technician”, “unemployed”, “unknown”)<br>
3.marital : marital status (categorical: “divorced”, “married”, “single”, “unknown”)<br>
4.education (categorical: “basic.4y”, “basic.6y”, “basic.9y”, “high.school”, “illiterate”, “professional.course”, “university.degree”, “unknown”)<br>
5.default: has credit in default? (categorical: “no”, “yes”, “unknown”)<br>
6.housing: has housing loan? (categorical: “no”, “yes”, “unknown”)<br>
7.loan: has personal loan? (categorical: “no”, “yes”, “unknown”)<br>
8.contact: contact communication type (categorical: “cellular”, “telephone”)<br>
9.month: last contact month of year (categorical: “jan”, “feb”, “mar”, …, “nov”, “dec”)<br>
10.day_of_week: last contact day of the week (categorical: “mon”, “tue”, “wed”, “thu”, “fri”)<br>
11.duration: last contact duration, in seconds (numeric). Important note: this attribute highly affects the output target (e.g., if duration=0 then y=’no’). The duration is not known before a call is performed, also, after the end of the call, y is obviously known. Thus, this input should only be included for benchmark purposes and should be discarded if the intention is to have a realistic predictive model <br>
12.campaign: number of contacts performed during this campaign and for this client (numeric, includes last contact)<br>
13.pdays: number of days that passed by after the client was last contacted from a previous campaign (numeric; 999 means client was not previously contacted)<br>
14.previous: number of contacts performed before this campaign and for this client (numeric)<br>
15.poutcome: outcome of the previous marketing campaign (categorical: “failure”, “nonexistent”, “success”)<br>
16.emp.var.rate: employment variation rate — (numeric)<br>
17.cons.price.idx: consumer price index — (numeric)<br>
18.cons.conf.idx: consumer confidence index — (numeric)<br>
19.euribor3m: euribor 3 month rate — (numeric)<br>
20.nr.employed: number of employees — (numeric)<br>

In [16]:
df["education"].unique()

array(['basic.4y', 'unknown', 'university.degree', 'high.school',
       'basic.9y', 'professional.course', 'basic.6y', 'illiterate'],
      dtype=object)

In [17]:
df['education'] = np.where(df['education'] == "basic.4y", 'basic', df['education'])
df['education'] = np.where(df['education'] == "basic.9y", 'basic', df['education'])
df['education'] = np.where(df['education'] == "basic.6y", 'basic', df['education'])
df['education'].unique()

array(['basic', 'unknown', 'university.degree', 'high.school',
       'professional.course', 'illiterate'], dtype=object)

In [19]:
df.dropna()

age          job  marital            education  default housing loan  \
0       44  blue-collar  married                basic  unknown     yes   no   
1       53   technician  married              unknown       no      no   no   
2       28   management   single    university.degree       no     yes   no   
3       39     services  married          high.school       no      no   no   
4       55      retired  married                basic       no     yes   no   
...    ...          ...      ...                  ...      ...     ...  ...   
41183   59      retired  married          high.school  unknown      no  yes   
41184   31    housemaid  married                basic  unknown      no   no   
41185   42       admin.   single    university.degree  unknown     yes  yes   
41186   48   technician  married  professional.course       no      no  yes   
41187   25      student   single          high.school       no      no   no   

         contact month day_of_week  ...  campaign  pdays  previous  \
0       cellular   aug         thu  ...         1    999         0   
1       cellular   nov         fri  ...         1    999         0   
2       cellular   jun         thu  ...         3      6         2   
3       cellular   apr         fri  ...         2    999         0   
4       cellular   aug         fri  ...         1      3         1   
...          ...   ...         ...  ...       ...    ...       ...   
41183  telephone   jun         thu  ...         1    999         0   
41184  telephone   may         thu  ...         2    999         0   
41185  telephone   may         wed  ...         3    999         0   
41186  telephone   oct         tue  ...         2    999         0   
41187  telephone   may         fri  ...         4    999         0   

          poutcome emp_var_rate  cons_price_idx  cons_conf_idx  euribor3m  \
0      nonexistent          1.4          93.444          -36.1      4.963   
1      nonexistent         -0.1          93.200          -42.0      4.021   
2          success         -1.7          94.055          -39.8      0.729   
3      nonexistent         -1.8          93.075          -47.1      1.405   
4          success         -2.9          92.201          -31.4      0.869   
...            ...          ...             ...            ...        ...   
41183  nonexistent          1.4          94.465          -41.8      4.866   
41184  nonexistent          1.1          93.994          -36.4      4.860   
41185  nonexistent          1.1          93.994          -36.4      4.857   
41186  nonexistent         -3.4          92.431          -26.9      0.742   
41187  nonexistent          1.1          93.994          -36.4      4.859   

       nr_employed  y  
0           5228.1  0  
1           5195.8  0  
2           4991.6  1  
3           5099.1  0  
4           5076.2  1  
...            ... ..  
41183       5228.1  0  
41184       5191.0  0  
41185       5191.0  0  
41186       5017.5  0  
41187       5191.0  0  

[41188 rows x 21 columns]

In [ ]:
df["material"].unique